qt/python/mantidqt/widgets/codeeditor/execution.py

```diff
@@ -90,15 +126,16 @@ def execute(self, code_str, filename=None):
             self.globals_ns[FILE_ATTR] = filename
         else:
             filename = EMPTY_FILENAME_ID
-        compile(code_str, filename, mode=COMPILE_MODE, dont_inherit=True)
+        flags = get_future_import_compiler_flags(code_str)
+        compile(code_str, filename, mode=COMPILE_MODE, dont_inherit=True, flags=flags)
```

Comment:
Looking at this again, is this line actually doing anything? As far as I can tell, it isn't.

Reply:
Ahh, it's checking for syntax errors in the whole script. When we don't have this line, the code will run up to the line with the syntax error. I'll add a comment so no one is tempted to remove it

Reply:
:+1: